bindings/python/tests/test_triplestore.py

```diff
@@ -103,89 +97,113 @@
 
-ex:Sum rdfs:subClassOf owl:Thing .
+    ex:sum_{fid}_output1 a fno:Output ;
+        map:mapsTo ex:Sum .
 
-ex:sum_{fid}_output1 a fno:Output ;
-    map:mapsTo ex:Sum .
+    ex:sum_{fid}_parameter1_a a fno:Parameter ;
+        rdfs:label "a"@en ;
+        map:mapsTo ex:MyConcept .
 
-ex:sum_{fid}_parameter1_a a fno:Parameter ;
-    rdfs:label "a"@en ;
-    map:mapsTo ex:MyConcept .
+    ex:sum_{fid}_parameter2_b a fno:Parameter ;
+        rdfs:label "b"@en ;
+        map:mapsTo ex:AnotherConcept .
 
-ex:sum_{fid}_parameter2_b a fno:Parameter ;
-    rdfs:label "b"@en ;
-    map:mapsTo ex:AnotherConcept .
+    ex:MyConcept rdfs:subClassOf owl:Thing .
 
-ex:MyConcept rdfs:subClassOf owl:Thing .
+    """)
+    assert s == expected
 
-"""
+    ts2 = Triplestore("rdflib")
+    ts2.parse(format="turtle", data=s)
+    assert ts2.serialize(format="turtle") == s
+    ts2.set((EX.AnotherConcept, RDFS.subClassOf, EX.MyConcept))
 
-ts2 = Triplestore("rdflib")
-ts2.parse(format="turtle", data=s)
-assert ts2.serialize(format="turtle") == s
-ts2.set((EX.AnotherConcept, RDFS.subClassOf, EX.MyConcept))
+    def cost(x):
+        return 2*x
 
-def cost(x):
-    return 2*x
+    ts2.add_mapsTo(EX.Sum, "http://onto-ns.com/meta/0.1/MyEntity#sum",
+                   cost=cost)
+    assert list(ts2.function_repo.values())[0] == cost
 
-ts2.add_mapsTo(EX.Sum, "http://onto-ns.com/meta/0.1/MyEntity#sum", cost=cost)
-assert list(ts2.function_repo.values())[0] == cost
+    def func(x):
+        return x+1
 
-def func(x):
-    return x+1
+    ts2.add_function(func, expects=EX.Sum, returns=EX.OneMore, cost=cost)
+    assert list(ts2.function_repo.values())[1] == func
+    assert len(ts2.function_repo) == 2  # cost is only added once
 
-ts2.add_function(func, expects=EX.Sum, returns=EX.OneMore, cost=cost)
-assert list(ts2.function_repo.values())[1] == func
-assert len(ts2.function_repo) == 2  # cost is only added once
+    def func2(x):
+        return x+2
 
-def func2(x):
-    return x+2
+    def cost2(x):
+        return 2*x+1
 
-def cost2(x):
-    return 2*x+1
+    ts2.add_function(func2, expects=EX.Sum, returns=EX.EvenMore, cost=cost2)
+    assert len(ts2.function_repo) == 4
 
-ts2.add_function(func2, expects=EX.Sum, returns=EX.EvenMore, cost=cost2)
-assert len(ts2.function_repo) == 4
+    #print(ts2.serialize(format="turtle"))
 
-#print(ts2.serialize(format="turtle"))
+
+# Test ontopy triplestore backend
+# -------------------------------
+if ontopy:
+    ts3 = Triplestore("ontopy", base_iri="emmo", load=True)
+    onto = ts3.backend.onto
+    triples = list(ts3.triples((None, None, None)))
+
+    ts4 = Triplestore(
+        "ontopy", base_iri="http://onto-ns.com/ontologies/examples/food",
+    )
+    ts4.parse(ontopath_food)
+
+    ts5 = Triplestore(
+        "ontopy", base_iri="http://onto-ns.com/ontologies/examples/food",
+    )
+    ts5.bind('food', FOOD)
+    with open(ontopath_food, 'rt') as f:
+        ts5.parse(data=f.read())
```

Comment:
Several functions that belong to the ontopy triplestore backend are not tested: 
add_triples, remove, query, triples, remove, serialize and update. 

These need to be added.

Reply:
We will add more tests in https://github.com/SINTEF/dlite/issues/341

Reply:
OK